src/brsTypes/components/RoSGNode.ts

```diff
@@ -526,48 +526,58 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
-            if (componentDef && functionname.value in componentDef.functions) {
-                return interpreter.inSubEnv((subInterpreter) => {
-                    let functionToCall = subInterpreter.getCallableFunction(functionname.value);
-                    if (!functionToCall) {
-                        interpreter.stderr.write(
-                            `Ignoring attempt to call non-implemented function ${functionname}`
-                        );
-                        return BrsInvalid.Instance;
+    private callfunc = new Callable(
+        "callfunc",
+        ...Callable.variadic({
+            signature: {
+                args: [new StdlibArgument("functionname", ValueKind.String)],
+                returns: ValueKind.Dynamic,
+            },
+            impl: (
+                interpreter: Interpreter,
+                functionname: BrsString,
+                ...functionargs: BrsType[]
+            ) => {
+                // We need to search the callee's environment for this function rather than the caller's.
+                let componentDef = interpreter.environment.nodeDefMap.get(this.nodeSubtype);
+
+                while (componentDef) {
+                    // Only allow public functions (defined in the interface) to be called.
+                    if (componentDef && functionname.value in componentDef.functions) {
+                        return interpreter.inSubEnv((subInterpreter) => {
+                            let functionToCall = subInterpreter.getCallableFunction(
+                                functionname.value
+                            );
+                            if (!functionToCall) {
+                                interpreter.stderr.write(
+                                    `Ignoring attempt to call non-implemented function ${functionname}`
+                                );
+                                return BrsInvalid.Instance;
+                            }
+
+                            subInterpreter.environment.setM(this.m);
+                            subInterpreter.environment.setRootM(this.m);
+
+                            // Determine whether the function should get arguments or not.
+                            if (functionToCall.getFirstSatisfiedSignature(functionargs)) {
+                                return functionToCall.call(subInterpreter, ...functionargs);
+                            } else {
+                                return functionToCall.call(subInterpreter);
+                            }
+                        }, componentDef.environment);
                     }
 
-                    subInterpreter.environment.setM(this.m);
-                    subInterpreter.environment.setRootM(this.m);
+                    // Try the parent node if we can't find an implementation here
+                    componentDef = interpreter.environment.nodeDefMap.get(componentDef.extends);
```

Comment:
Did we not merge those functions down into child classes? 🤔

Reply:
> Makes sense to me! I'm a little surprised that we didn't flatten all the <function> definitions when we flatted the rest of the <interface> — do you happen to know if there's a reason we didn't?

Great question. I remember having this conversation before. I think what we landed on was that we are actually _not_ flattening our component definitions. Instead, when we instantiate them in `createNodeByType`, we traverse the inheritance tree. I know @alimnios72 filed #408 a while ago that touched on this.

Actually, if you look at the [component definition flattening code](https://github.com/sjbarag/brs/blob/master/src/componentprocessor/index.ts#L128-L151), we're just ignoring `inheritedFields` completely, which was referenced [in this PR for on-change callbacks](https://github.com/sjbarag/brs/pull/424#discussion_r432173107). I know there was a reason for doing that; I am struggling to remember exactly why. It was something to do with the on-change callbacks not existing in the scope of a child component when the definitions were flattened maybe?

Reply:
I think we only for `init` methods but never for functions available in the component interface. We should probably do in the future to avoid searching the correct function at runtime